src/algorithms/tonal/audio2pitch.cpp

```diff
@@ -0,0 +1,130 @@
+#include "audio2pitch.h"
+
+using namespace essentia;
+using namespace standard;
+
+const char* Audio2Pitch::name = "Audio2Pitch";
+const char* Audio2Pitch::category = "Pitch";
+const char* Audio2Pitch::description = DOC("Extractor algorithm to compute pitch with several possible pitch algorithms, specifically targeted for real-time pitch detection on saxophone signals.");
+
+bool Audio2Pitch::isAboveThresholds(Real pitchConfidence, Real loudness) {
+  return (pitchConfidence >= _pitchConfidenceThreshold) && (loudness >= _loudnessThreshold);
+}
+
+void Audio2Pitch::configure() {
+
+  _sampleRate = parameter("sampleRate").toReal();
+  _frameSize = parameter("frameSize").toInt();
+  _minFrequency = parameter("minFrequency").toReal();
+  _maxFrequency = parameter("maxFrequency").toReal();
+  _pitchAlgorithmName = parameter("pitchAlgorithm").toString();
+  _loudnessAlgorithmName = parameter("loudnessAlgorithm").toString();
+  _tolerance = parameter("tolerance").toReal();
+  _pitchConfidenceThreshold = parameter("pitchConfidenceThreshold").toReal();
+  _loudnessThreshold = parameter("loudnessThreshold").toReal();
+
+  if (_maxFrequency > _sampleRate*0.5 ) {
+    throw EssentiaException("Audio2Pitch: Max frequency cannot be higher than Nyquist frequency");
+  }
+  if (_maxFrequency <= _minFrequency) {
+    throw EssentiaException("Audio2Pitch: Max frequency cannot be lower than min frequency");
+  }
+
+  if (_pitchAlgorithmName != "pyin_fft" && _pitchAlgorithmName != "pyin") {
+    E_INFO("Audio2Pitch: 'pitchAlgorithm' = "<<_pitchAlgorithmName<<"\n");
```

Comment:
Why not to add the parameter value directly to the exception message?

Reply:
mmm... because operands are invalid in the exception message:
`../src/algorithms/tonal/audio2pitch.cpp:60:69: error: invalid operands to binary expression ('const char [39]' and 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >'))
    throw EssentiaException("Audio2Pitch: Bad 'loudnessAlgorithm' ="<<_loudnessAlgorithmName<<"\n");`
  Can you confirm that?

Reply:
ok, I have seen how you use to do that in some other algorithm like here:
`throw EssentiaException(fullName(), ": Could not push 1 value, output buffer is full");`

Reply:
mmmm... interesting there is not E_INFO message in my last commits. Maybe do you need to pull my last changes?